### PR TITLE
47015 core data entity attribute relationship redesign modified

### PR DESCRIPTION
**Pull request by Tahia50**

### CODE REVIEW

File: hrm_2/AppDelegate.m [hunk 1]
```diff
@@ -42,4 +42,87 @@ - (void)applicationWillTerminate:(UIApplication *)application {
...
+
+@synthesize managedObjectContext = _managedObjectContext;
+@synthesize managedObjectModel = _managedObjectModel;
+@synthesize persistentStoreCoordinator = _persistentStoreCoordinator;
+
+- (NSURL *)applicationDocumentsDirectory {
+    // The directory the application uses to store the Core Data store file. This code uses a directory named "xyz.ddd" in the application's documents directory.
+    return [[[NSFileManager defaultManager] URLsForDirectory:NSDocumentDirectory inDomains:NSUserDomainMask] lastObject];
+
+
```
Comment (nguyenvandungb) [inline hrm_2/AppDelegate.m]: Please do not plus many space like this to make file code clearly

File: hrm_2/CoreData/Department+CoreDataProperties.h [hunk 2]
```diff
@@ -15,8 +15,17 @@ NS_ASSUME_NONNULL_BEGIN
 
 @interface Department (CoreDataProperties)
 
-@property (nullable, nonatomic, retain) NSNumber *departmant_id;
```
Comment (nguyenvandungb) [inline hrm_2/CoreData/Department+CoreDataProperties.h]: Don't need this property any more?. I think you need it when you get data from server via api.

Comment (Tahia50) [inline hrm_2/CoreData/Department+CoreDataProperties.h]: should n't department name be enough?? names are suppose to be unique...right??

Comment (nguyenvandungb) [inline hrm_2/CoreData/Department+CoreDataProperties.h]: It maybe unique but better if we have id. But we can update if need later. I will accept this pull for now.

**Pull request merged**